core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -470,78 +460,87 @@ protected void doSample() {
 
             final CountDownLatch latch = new CountDownLatch(nodesToPing.size());
             final ConcurrentMap<DiscoveryNode, ClusterStateResponse> clusterStateResponses = ConcurrentCollections.newConcurrentMap();
-            for (final DiscoveryNode listedNode : nodesToPing) {
-                threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(new Runnable() {
-                    @Override
-                    public void run() {
-                        try {
-                            if (!transportService.nodeConnected(listedNode)) {
-                                try {
+            try {
+                for (final DiscoveryNode nodeToPing : nodesToPing) {
+                    threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(new AbstractRunnable() {
+
+                        /**
+                         * we try to reuse existing connections but if needed we will open a temporary connection
+                         * that nodes to be closed at the end of execution.
```

Comment:
`s/that nodes to be closed at the end of execution./to a node that will be closed at the end of the excution`?